src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -1018,12 +1030,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
-        Private ReadOnly Property IConversionExpression_IsExplicit As Boolean Implements IConversionExpression.IsExplicit
+        Private ReadOnly Property IConversionExpression_ExplicitCastInSource As Boolean Implements IConversionExpression.ExplicitCastInSource
             Get
                 Return Me.ExplicitCastInCode
             End Get
         End Property
 
+        Private ReadOnly Property IConversionExpression_Conversion As IConversion Implements IConversionExpression.Conversion
+            Get
+                Return New Conversion(Me.ConversionKind, method:=Nothing)
```

Comment:
I don't understand the reasoning here.  It sounds like the implementation is *already* incorrect (as per #8872 ).  It seems like when that issue is fixed we can address the concern you have here with this.  It doesn't make sense to me that we would have to gate fleshing out the API appropriately with also needing to fix other pre-existing issues.

As long as this is properly tracked, it seems like it would be not an issue to get this in.  @srivatsn / @JohnHamby  What do you think?